lib/Cron/Workers/StatsNotifications/StatsNotificationsRepository.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace MailPoet\Cron\Workers\StatsNotifications;
+
+use MailPoet\Doctrine\Repository;
+use MailPoet\Entities\StatsNotificationEntity;
+
+class StatsNotificationsRepository extends Repository {
+  protected function getEntityClassName() {
+    return StatsNotificationEntity::class;
+  }
+
+  /**
+   * @param int $newsletter_id
+   * @return array
+   */
+  public function findAllForNewsletter($newsletter_id) {
+    return $this->doctrine_repository
+      ->createQueryBuilder('stn')
+      ->join('stn.task', 'tasks')
+      ->join('stn.newsletter', 'n')
```

Comment:
I am a bit confused here. The `Worker` class that is calling this method does need the newsletter for its data. That is the main reason I added it here, to load it and prevent another DB query since I can load everything at once. You are saying the JOIN is not needed. Why is that?

Reply:
If I'm reading the code correctly the current implementation calls it only like this:
```
private function hasTaskBeenScheduled($newsletter_id) {
    $existing = $this->repository->findByNewsletterId($newsletter_id);
    return $existing instanceof StatsNotificationEntity;
  }
```
Or can you send me a link ot a different usage?